File: config/libIndex.js
// 拷贝文件
const path = require('path');
const fs = require('fs');
const { resolve } = path;

// 获取时间
const TimeFn = require('../get_time');
const {
    name, version, author, license
} = require('../package.json');
const banner = `/**
* @${name} v${version}
* (c) 2020-2021 ${author}
* Released under the ${license} License.
* ${TimeFn()}
*/
`;

// 写入index.d.ts
const copyFile = (url, out) => {
    fs.readFile(resolve(__dirname, url), (err, data) => {
        fs.writeFile(resolve(__dirname, out), (banner + data), err => {
            if (err) {
                console.log(`${out}写入出现错误 ${err.toString()}`);
            } else {
                console.log(`${out}完成!`);
            }
        })
    })
};
copyFile('../src/components/index.ts', '../lib/index.d.ts');

const firstToUpperCase = (str) => {
    const arr = str.split('');
    arr[0] = arr[0].toUpperCase();
    return arr.join('');
}
// 写入index.js index.d.ts
const readFile = (url) => {
    fs.readdir(resolve(__dirname, url), (err, data) => {
        const arr = data.filter(d => !d.includes('.'));
        const resultJs = [], resultTs = [];
        arr.forEach(d => {
            resultJs.push(`export { default as ${firstToUpperCase(d)} } from './${d}';`);
            resultTs.push(`export type { Props as ${firstToUpperCase(d)}Props } from './${d}';\nexport { default as ${firstToUpperCase(d)} } from './${d}';`);
        })
        const resJs = `\n${resultJs.join('\n')}\n`;
        fs.writeFile(resolve(__dirname, `${url}/index.js`), (banner + resJs), err => {
            if (err) {
                console.log(`js写入出现错误 ${err.toString()}`);
            } else {
                console.log('js完成!');
            }
        });
        const resTs = `\n${resultTs.join('\n')}\n`;
        /*fs.writeFile(resolve(__dirname, `${url}/index.d.ts`), (banner + resTs), err => {
            if (err) {
                console.log(`ts写入出现错误 ${err.toString()}`);
            } else {
                console.log('ts完成!');
            }
        });*/
    });
};
readFile('../lib');

File: config/libIndex.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"os"
	"path/filepath"
	"time"
)

type Info struct {
	Name    string `json:"name"`
	Version string `json:"version"`
	Author  string `json:"author"`
	License string `json:"license"`
}

func main() {
	pwd, _ := os.Getwd()

	banner := getBanner(pwd)

	files, errReadDir := os.ReadDir("lib")
	if errReadDir != nil {
		log.Fatal("-----", errReadDir)
	}

	var jsImport string

	for _, file := range files {
		fileName := file.Name()
		jsImport += fmt.Sprintf("export { default as %v } from './%v';\n", fileName, fileName)
	}
	js := banner + jsImport

	errJsWrite := os.WriteFile(filepath.Join(pwd, "lib/index.js"), []byte(js), 0666)
	if errJsWrite != nil {
		log.Fatal("js写入错误-----", errJsWrite)
	}
	copyFile(pwd)
}

func getBanner(pwd string) string {
	YEAR := time.Now().Year()
	var info Info
	jsonFile, errJson := os.Open(filepath.Join(pwd, "package.json"))
	if errJson != nil {
		log.Fatal("errJson-----", errJson)
	}
	defer jsonFile.Close()
	decoder := json.NewDecoder(jsonFile)
	errDecode := decoder.Decode(&info)
	if errDecode != nil {
		log.Fatal("errDecode-----", errDecode)
	}
	return fmt.Sprintf("/**\n* @%v v%v\n* (c) 2019-%v %v\n* Released under the %v License.\n* %v\n*/\n", info.Name, info.Version, YEAR, info.Author, info.License, time.Now().Format("2006.01.02 15:04:05"))
}

func copyFile(pwd string) {
	data, errRead := os.ReadFile(filepath.Join(pwd, "src/components/index.ts"))
	if errRead != nil {
		log.Fatal("-----", errRead)
	}
	errWrite := os.WriteFile(filepath.Join(pwd, "lib/index.d.ts"), data, 0666)
	if errWrite != nil {
		log.Fatal("-----", errWrite)
	}
}
